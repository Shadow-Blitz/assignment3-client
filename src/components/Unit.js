import React, { Component } from 'react';
import { Fab, Paper } from '@material-ui/core/';
import AddIcon from '@material-ui/icons/Add/';
import { api, setJwt } from '../api/init'
import { Link } from 'react-router-dom';
import Program from './Program'

const style = {
  Paper: {
      'width': '400px',
      'margin': '10% auto 0 auto',
      'textAlign': 'center',
      'padding': '5%'

  }
}

class Unit extends Component {
  state = {
    users: [],
    programs: []
  }

  componentDidMount() {
    const token = localStorage.getItem('token')
    setJwt(token)
    api.get('/units/myUnit').then((res) => {

      this.setState({ ...res.data })

    }).catch((err) => {
      console.error('Could not fetch user', err)
    })
  }
  

  render() {
    console.log(this.state.users[0])
    console.log(this.state.programs, 'programs')
    return (
      <React.Fragment>
        <h1>{this.state.unit && this.state.unit.name}</h1>
        <Paper style={style.Paper}>
          <h2>Guide Hut</h2>
        </Paper>

        {/* Display a list of leaders associated with the unit */}
        <Paper style={style.Paper}>
          <h2>Unit Leaders</h2>
          {this.state.users.map((user) => <p key={user._id}>{user.name.firstname} {user.name.lastname}</p>)}
        </Paper>

        {/* Display a list of programs created by the unit leaders */}
        <Paper style={style.Paper}>
          <h2>Unit Programs</h2>
          {this.state.programs.map((program) => {
              return (
                <Program key={program._id} program={program}></Program>
              )
            })}
          <Link to='/create-program'>
            <Fab size="medium" color="secondary" aria-label="Add" style={{ 'backgroundColor': 'orange' }}>
              <AddIcon />
            </Fab>
          </Link>
        </Paper>

        {/* Display a list of members belonging to the unit */}
        <Paper style={style.Paper}>
          <h2>Unit Members</h2>
          <Fab className="plusButton" size="medium" color="secondary" aria-label="Add" style={{ 'backgroundColor': 'orange' }}>
            <AddIcon />
          </Fab>
        </Paper>
      </React.Fragment>
    )
  }
}
export default Unit;