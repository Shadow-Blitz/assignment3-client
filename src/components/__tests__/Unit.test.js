import React from 'react';
import Unit from '../Unit';
import { shallow } from 'enzyme';

describe("Unit", () => {
    it("should render correctly", () => {
        const component = shallow(<Unit />);
    });
    it("should render initial layout", () => {
        // when
        const component = shallow(<Unit />);
        // then
        expect(component.getElements()).toMatchSnapshot();
    });
})