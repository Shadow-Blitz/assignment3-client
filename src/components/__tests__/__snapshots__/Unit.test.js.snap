// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit should render initial layout 1`] = `
Array [
  <React.Fragment>
    <h1 />
    <h2>
      Guide Hut
    </h2>
    <WithStyles(Divider) />
    <h2>
      Unit Leaders
    </h2>
    <WithStyles(Grid)
      alignItems="center"
      container={true}
      direction="row"
      justify="center"
      spacing={16}
    />
    <WithStyles(Divider) />
    <h2>
      Unit Programs
    </h2>
    <WithStyles(Grid)
      alignItems="center"
      container={true}
      direction="row"
      justify="center"
      spacing={16}
    />
    <Link
      replace={false}
      to="/create-program"
    >
      <WithStyles(Fab)
        aria-label="Add"
        color="secondary"
        size="medium"
        style={
          Object {
            "backgroundColor": "orange",
            "margin": 15,
          }
        }
      >
        <pure(AddIcon) />
      </WithStyles(Fab)>
    </Link>
    <WithStyles(Divider) />
    <h2>
      Unit Members
    </h2>
    <WithStyles(Fab)
      aria-label="Add"
      className="plusButton"
      color="secondary"
      size="medium"
      style={
        Object {
          "backgroundColor": "orange",
          "margin": 15,
        }
      }
    >
      <pure(AddIcon) />
    </WithStyles(Fab)>
  </React.Fragment>,
]
`;
